int32 distance
---
int32 total_distance
---
int32 traveled

